js/bitmex.js

```diff
@@ -190,11 +190,28 @@ module.exports = class bitmex extends Exchange {
             };
             const lotSize = this.safeFloat (market, 'lotSize');
             const tickSize = this.safeFloat (market, 'tickSize');
+            // price_to_precision/amount_to_precision has no understanding of ticksize
+            // if the tickSize is 0.2, 0.1 has the same precision, but gets
+            // rejected by the API.
+            // So we need to reduce the precision to create valid prices/amounts
```

Comment:
The reasoning behind implementing the `precision + 1` logic (as it is) instead of `precision - 1` (as in these lines) was to still provide the user with all the range of values, if the user handles precision in userland. That's mostly for historical reason, and, I think, a better approach would be to generalize non-decimal precision or _lot sizes_. Let me know what you think about that.

Reply:
@kroitor 
Yes, I know that it reduces the range of values, that's why I also proposed the solution to add `TICK_SIZE` as precision option and I totally agree that we should generalize precision.
`DECIMAL_PLACES` can be translated to `TICK_SIZE`, the other way around is not possible.

I thought, the patch would fix the issue temporary, but I havent thought of users handling the precision in userland, valid point.

Reply:
@brandsimon how do we proceed on this? Close this for now, then add a corresponding precision mode first, then add the new handling code to BitMEX? I'm afraid, if I merge it as is, it will be backward-incompatible, and sometimes those tiny fractions matter...

Reply:
@kroitor Yes, I totally agree. Merging is not a good solution.
Regarding the corresponding precision mode, are there any guidelines? (Rounding up/down, pitfalls and so on?)

Reply:
@brandsimon 

> Regarding the corresponding precision mode, are there any guidelines? (Rounding up/down, pitfalls and so on?)

Not yet, but we've had a corresponding request from @mkutny earlier here: https://github.com/ccxt/ccxt/issues/2733.

In general, prices are often rounded arithmetically, amounts are often truncated.